lightning/src/util/events.rs

```diff
@@ -352,18 +343,17 @@ impl Writeable for Event {
 			&Event::PaymentReceived { ref payment_hash, ref amt, ref purpose } => {
 				1u8.write(writer)?;
 				let mut payment_secret = None;
-				let mut user_payment_id = None;
 				let payment_preimage;
 				match &purpose {
-					PaymentPurpose::InvoicePayment { payment_preimage: preimage, payment_secret: secret, user_payment_id: id } => {
+					PaymentPurpose::InvoicePayment { payment_preimage: preimage, payment_secret: secret } => {
 						payment_secret = Some(secret);
 						payment_preimage = *preimage;
-						user_payment_id = Some(id);
 					},
 					PaymentPurpose::SpontaneousPayment(preimage) => {
 						payment_preimage = Some(*preimage);
 					}
 				}
+				let user_payment_id: Option<u64> = None; // for compatibility with 0.0.103 and earlier
```

Comment:
We need to write `Some(0)` here - looks like the old code will return `InvalidValue` if it's `None`, which we don't want.